src/lib/http.coffee

```diff
@@ -58,25 +51,33 @@ module.exports = (dnschain) ->
                 res.end()
                 cb()
 
-            resolver = switch req.headers.host
-                when 'icann.dns' then 'dns'
-                when 'namecoin.dns' then 'nmc'
-                when 'bitshares.dns' then 'bdns'
+            [...,resolverName] =
+                if S(req.headers.host).endsWith('.dns')
+                    S(req.headers.host).chompRight('.dns').s.split('.')
+                else if S(req.headers.blockchain).endsWith('.dns')
+                    S(req.headers.blockchain).chompRight('.dns').s.split('.')
                 else
-                    @log.warn gLineInfo "unknown host type: #{req.headers.host} -- defaulting to namecoin.dns!"
-                    'nmc'
+                    ['none']
```

Comment:
I think this can be simplified to:

```coffeescript
            [...,resolverName] =
                if S(header = req.headers.blockchain ? req.headers.host).endsWith('.dns')
                    S(req.headers[header]).chompRight('.dns').s.split('.')
                else
                    ['none']
```